internal/controller/bmcsettings_controller.go

```diff
@@ -495,26 +499,41 @@ func (r *BMCSettingsReconciler) checkIfMaintenanceGranted(
 	log logr.Logger,
 	bmcSetting *metalv1alpha1.BMCSettings,
 	bmcClient bmc.BMC,
-) bool {
+) (bool, metalv1alpha1.ServerMaintenanceState) {
```

Comment:
It seems this function returns one of three states (`Unknown`, `SomeMaintenanceMissing`, `AllInMaintenance`). Could these be returned as a single instance of an enum?